### PR TITLE
Cosmology branch

### PR DESCRIPTION
**Pull request by Gustavobflh**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: physics/hubble_parameter.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    z: redshift
+
+    Returns
+    -------
+    result : Hubble parameter in and the unit km/s/Mpc (the unit can be changed
+             if you want)
+
+    >>> hubble_parameter(H_0=63.3, Omg_r=1e-4, Omg_m=-0.3, Omg_l=0.7, z=1)
+    ...
+    ValueError: Any input parameter can be negative
```
Comment (CaedenPH) [inline physics/hubble_parameter.py:48]:
```suggestion
    >>> hubble_parameter(H_0=63.3, Omg_r=1e-4, Omg_m=-0.3, Omg_l=0.7, z=1)
    Traceback(most recent call last):
        ...
    ValueError: Any input parameter can be negative
```

File: physics/hubble_parameter.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    """
+
+    if z < 0 or Omg_m < 0 or Omg_r < 0 or Omg_l < 0:
+        raise ValueError("Any input parameter can be negative")
+    if Omg_m > 1 or Omg_r > 1 or Omg_l > 1:
+        raise ValueError("Relative densities cannot be greater than one")
+    else:
+        Omg_k = 1 - (Omg_m + Omg_r + Omg_l)
+        E2 = Omg_r * (z + 1) ** 4 + Omg_m * (z + 1) ** 3 + Omg_k * (z + 1) ** 2 + Omg_l
+        H = H_0 * E2 ** (1 / 2)
```
Comment (CaedenPH) [inline physics/hubble_parameter.py:62]:
```suggestion
        H = H_0 * E2 ** 0.5
```

File: physics/hubble_parameter.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    ValueError: Any input parameter can be negative
+
+     >>> hubble_parameter(H_0=63.3, Omg_r=1e-4, Omg_m= 1.2, Omg_l=0.7, z=1)
+     ...
+     ValueError: Relative densities cannot be greater than one
+    """
+
+    if z < 0 or Omg_m < 0 or Omg_r < 0 or Omg_l < 0:
+        raise ValueError("Any input parameter can be negative")
+    if Omg_m > 1 or Omg_r > 1 or Omg_l > 1:
```
Comment (CaedenPH) [inline physics/hubble_parameter.py:57]:
```suggestion
    if any(p > 1 for p in (Omg_m, Omg_r, Omg_l)):
```

File: physics/hubble_parameter.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+of the densities by the completness
+
+Omg_k = 1 - (Omg_m + Omg_r + Omg_l)
+
+Source :
+https://www.sciencedirect.com/topics/mathematics/hubble-parameter
+"""
+
+
+def hubble_parameter(H_0: float, Omg_r: float, Omg_m: float, Omg_l: float, z: float):
```
Comment (cclauss) [inline physics/hubble_parameter.py:25]: Variable names should be words that humans can understand without looking at the comments.
```suggestion
def hubble_parameter(
    hubble_constant: float,
    radiation_density: float,
    mass_density: float,
    dark_energy_density: float,
    redshift: float,
) -> float:
```

**Comment by cclauss** [comment]
Please run this on your own computer:
`python3 -m doctest -v physics/hubble_parameter.py`

**Comment by cclauss** [comment]
physics/hubble_parameter.py:25:23: N803 argument name 'H_0' should be lowercase
physics/hubble_parameter.py:60:10: N806 variable 'Omg_k' in function should be lowercase
physics/hubble_parameter.py:61:10: N806 variable 'E2' in function should be lowercase
physics/hubble_parameter.py:62:10: N806 variable 'H' in function should be lowercase

mypy.....................................................................Passed
codespell................................................................Failed
- hook id: codespell
- exit code: 65

physics/hubble_parameter.py:5: costumary ==> customary
physics/hubble_parameter.py:16: completness ==> completeness
physics/hubble_parameter.py:72: aproximation ==> approximation

File: physics/hubble_parameter.py (hunk 4 above)
Comment (cclauss) [reply physics/hubble_parameter.py:25]: Do this

**Comment by cclauss** [comment]
[CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md) was not read.  Please do not open new pull requests until November 1st. 2022.

**Pull request closed without merging**